### ISSUE TITLE
[tracking] Falco v0.33 release

### ISSUE THREAD

**Issue by jasondellaluce** [falcosecurity/falco#2230]
## Scheduled to happen: ~2022-09-30 (delayed)

**POSTPONED**
~_ETA: ~one week after falcosecurity/libs v0.9.0 release_~ **Released on 2022-10-19**

During the community call of 2022-09-28 I proposed myself as a release manager, and together with the other maintainers we plan to release v0.33 as soon as possible. Different factors caused this release to be running late, but this was mostly due to the new experimental release process, refactoring and fixing some parts of falcosecurity/libs, and lack of bandwidth from the maintainers.

Will keep this issue updated with the current status and progress.

## Current Status

- [x] Code freeze in falcosecurity/libs
- [x] Code taw in falcosecurity/libs
- [x] Release libs v0.9.0: https://github.com/falcosecurity/libs/releases/tag/0.9.0
- [x] Release drivers v3.0.1: https://github.com/falcosecurity/libs/releases/tag/3.0.1%2Bdriver
- [x] Code freeze in falcosecurity/falco
- [x] Open release branch in Falco
  - [x] https://github.com/falcosecurity/falco/tree/release/0.33.0
  - [x] Branch-protect the branch: https://github.com/falcosecurity/test-infra/pull/871
- [x] Code thaw in falcosecurity/falco
- [x] Release Falco v0.33.0

## Release Steps

Process is described in [this document](https://github.com/falcosecurity/falco/blob/master/RELEASE.md), but we are experimenting the new process drafted in https://github.com/falcosecurity/libs/pull/580.

## Release Blocking PRs

- [x] https://github.com/falcosecurity/falco/pull/2138
- [x] ~https://github.com/falcosecurity/falco/pull/2202~ _(postponed to next release)_
- [x] https://github.com/falcosecurity/falco/pull/2232
- [x] https://github.com/falcosecurity/falco/pull/2234
- [x] https://github.com/falcosecurity/falco/pull/2235
- [x] https://github.com/falcosecurity/falco/pull/2236
- [x] https://github.com/falcosecurity/falco/pull/2237
- [x] https://github.com/falcosecurity/falco/pull/2239
- [x] https://github.com/falcosecurity/falco/pull/2238
- [x] https://github.com/falcosecurity/falco/pull/2243
- [x] https://github.com/falcosecurity/falco/pull/2244
- [x] https://github.com/falcosecurity/falco/pull/2247
- [x] https://github.com/falcosecurity/falco/pull/2248
- [x] https://github.com/falcosecurity/falco/pull/2249
- [x] https://github.com/falcosecurity/falco/pull/2252
- [x] https://github.com/falcosecurity/falco/pull/2253
- [x] https://github.com/falcosecurity/falco/pull/2255
- [x] https://github.com/falcosecurity/falco/pull/2256
- [x] https://github.com/falcosecurity/falco/pull/2257
- [x] ~https://github.com/falcosecurity/falco/pull/2258~ _(postponed to next release)_
- [x] https://github.com/falcosecurity/libs/pull/653
- [x] https://github.com/falcosecurity/libs/pull/652
- [x] https://github.com/falcosecurity/libs/pull/655
- [x] https://github.com/falcosecurity/libs/pull/656
- [x] https://github.com/falcosecurity/libs/pull/660
- [x] https://github.com/falcosecurity/libs/pull/661
- [x] https://github.com/falcosecurity/libs/pull/662
- [x] https://github.com/falcosecurity/libs/pull/663
- [x] https://github.com/falcosecurity/libs/pull/665
- [x] https://github.com/falcosecurity/libs/pull/669

## Manual Testing Action Items

- [x] Running Falco on Kubernetes with the official Helm Charts
- [x] Running Falco from RPM and DEB artifacts
- [x] Running Falco in a container with the official images
- [x] Running/fuzzing Falco with multiple event sources active in parallel
- [x] Running/fuzzing Falco with variable syscall buffer dimension
- [x] Running Falco in all officially-supported architectures (x86_64, ARM64)
- [x] Running Falco with the supported drivers (kmod, eBPF, gVisor, trace files)
- [x] Test Falco with event generator
- [x] Test that plugins are correctly loaded
- [x] Test memory and CPU usage (with Valgrind and other tools)
- [x] Test latest version of driver loader
- [x] Test that k8s metadata enrichment client works as expected
- [x] Test that all Falco CLI options work as expected
- [x] Check that Falco log messages are correct and consistent
- [x] Test that Falco ruleset loading and validation works as expected

## Action Items

- [x] **Pre-Release**
  - [x]  Milestone: https://github.com/falcosecurity/falco/milestone/24
  - [x] Sync with new falcosecurity/libs release v0.9.0:
    - [x] https://github.com/falcosecurity/libs/issues/626
    - [x] https://github.com/falcosecurity/falco/pull/2257
  - [x] Sync with new falcosecurity/libs driver release v3.0.1:
    - [x] https://github.com/falcosecurity/falco/pull/2251
  - [x] Release plugin-sdk-go: https://github.com/falcosecurity/plugin-sdk-go/releases/tag/v0.6.0
  - [x] Release plugins:
    - [x] https://github.com/falcosecurity/plugins/pull/168 
    - [x] https://github.com/falcosecurity/plugins/releases
  - [x] Prebuilt drivers publish
    - https://github.com/falcosecurity/test-infra/pull/869
  - [x] Changelog
    - [x] https://github.com/falcosecurity/falco/pull/2261
    - [x] release branch sync: https://github.com/falcosecurity/falco/pull/2264
- [x] **Release**
  - [x] Tag: https://github.com/falcosecurity/falco/tree/0.33.0
  - [x] Github Release: https://github.com/falcosecurity/falco/releases/tag/0.33.0
- [x] **Website**
  - [x] Create 0.32 snapshot
    - [x] Protect snapshot branch: https://github.com/falcosecurity/test-infra/pull/876
  - [x] https://github.com/falcosecurity/falco-website/pull/699
  - [x] https://github.com/falcosecurity/falco-website/pull/659
  - [x] https://github.com/falcosecurity/falco-website/pull/701
  - [x] https://github.com/falcosecurity/falco-website/pull/702
  - [x] https://github.com/falcosecurity/falco-website/pull/703
  - [x] https://github.com/falcosecurity/falco-website/pull/704
  - [x] https://github.com/falcosecurity/falco-website/pull/705
  - [x] https://github.com/falcosecurity/falco-website/pull/706
  - [x] https://github.com/falcosecurity/falco-website/pull/708
  - [x] https://github.com/falcosecurity/falco-website/pull/710
  - [x] https://github.com/falcosecurity/falco-website/pull/714
  - [x] https://github.com/falcosecurity/falco-website/pull/716
- [x] **Announcements**
  - [x] mailing list 
  - [x] slack channel: https://kubernetes.slack.com/archives/CMWH3EH32/p1666194408614249
- [x] **Post-Release**
  - [x] Update helm charts
    - [x] https://github.com/falcosecurity/charts/pull/409
  - [x] Archive community call meeting notes:
    - [x] https://github.com/falcosecurity/community/pull/161

## Miscs

/milestone 0.33.0

/assign @jasondellaluce 

/kind documentation

**Comment by jasondellaluce** [comment]
Code freeze on Falco starts today!

**Comment by jasondellaluce** [comment]
Release branch has been opened! https://github.com/falcosecurity/falco/tree/release/0.33.0

**Comment by leogr** [comment]
Not actually an item for the release, but I want to keep a note of it here, so we won't forget.
After the release, we need to release the event-generator too, because of this :point_down: 
https://github.com/falcosecurity/event-generator/pull/77

**Comment by leogr** [comment]
For the same reason 
:point_down: 
https://github.com/falcosecurity/falco-exporter/issues/74

**Comment by jasondellaluce** [comment]
Code taw for Falco happens now! We can resume regular PR merging on mainline!

**Comment by jasondellaluce** [comment]
Falco has been released! 🥳

**Comment by leogr** [comment]
`event-generator`, `falco-exporter` and all charts have been updated!

I think it's time to close this issue, cc @jasondellaluce

**Comment by Andreagit97** [comment]
A big thank you to @alacuku for the help with helm charts!
